Task finished

LesterOper opened the pull request:
Task4 was finished
Implemented CartPageServlet, DeleteItemFromCartServlet
Implemented deleting from cart, updating cart, jsp "cartPage" and minicart.

LesterOper committed:
Task finished

Task4 was finished
Implemented CartPageServlet, DeleteItemFromCartServlet
Implemented deleting from cart, updating cart, jsp "cartPage" and minicart.

SergeiStarikov commented on src/main/java/com/es/phoneshop/Cart/Cart.java:
```diff
@@ -1,16 +1,32 @@
+        for (int i = 0; i < cartItems.size(); i++) {
+            summary = summary.add(cartItems.get(i).getTotalCost());
+        }
+        totalCartCost = summary;
```
Can we use streams approach for this?

SergeiStarikov commented on src/main/java/com/es/phoneshop/Cart/HttpSessionCartService.java:
```diff
@@ -1,72 +1,91 @@
+            cartList.getList().stream().filter(prod -> prod.getProduct().getId().equals(product.getId()))
+                    .findFirst()
+                    .get().setTotalCost(product.getPrice(), quantity);
+            cartList.setTotalCartCost(cartList.getList());
```
Please try to reuse the duplicate code.

SergeiStarikov commented on src/main/java/com/es/phoneshop/web/CartPageServlet.java:
```diff
@@ -0,0 +1,73 @@
+    public void showPage(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+        //request.setAttribute("products",
+        //      product.getProduct(Long.parseLong(request.getRequestURI().substring(request.getRequestURI().lastIndexOf("/") + 1))));
+        request.getRequestDispatcher("/WEB-INF/pages/cartPage.jsp").
```
Remove commented code.

SergeiStarikov commented on src/main/java/com/es/phoneshop/web/DeleteItemFromCartServlet.java:
```diff
@@ -0,0 +1,37 @@
+        while (cartItem.hasNext()) {
+            CartItem item = cartItem.next();
+            if (item.getProduct().getId() == product.getId()) {
+                BigDecimal cost = item.getTotalCost();
```
Remove unused variable.

SergeiStarikov commented on src/main/webapp/WEB-INF/pages/cartPage.jsp:
```diff
@@ -0,0 +1,60 @@
+        <c:set var="error" value="${requestScope['errors']}"/>
+        <form id="deleteCartItem" action="${pageContext.servletContext.contextPath}/products/cart/deleteItem"
+              method="post">
+            <c:forEach items="${cart}" var="cartItem" varStatus="status">
```
What if there are no items in the cart? Will we get table header with empty rows?
It's worth checking for empty cart before rendering the table.

LesterOper committed:
Fixed according comments

Fixed cart page, reused the dublicate code in method addProduct in class HttpSessionCartService
In class Cart, method "setTotalCartCost", fixed without for

SergeiStarikov commented on src/main/java/com/es/phoneshop/Cart/Cart.java:
```diff
@@ -1,7 +1,5 @@
 package com.es.phoneshop.Cart;
```
Please add more tests for boundary cases.
